java/pom.xml

```diff
@@ -271,55 +273,70 @@
         </configuration>
       </plugin>
 
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-checkstyle-plugin</artifactId>
+        <version>2.17</version>
+        <dependencies>
+          <dependency>
+            <groupId>com.puppycrawl.tools</groupId>
+            <artifactId>checkstyle</artifactId>
+            <version>6.19</version>
+          </dependency>
+          <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>${dep.guava.version}</version>
+          </dependency>
+          <dependency>
+              <groupId>org.slf4j</groupId>
+              <artifactId>jcl-over-slf4j</artifactId>
+              <version>1.7.5</version>
+          </dependency>
+        </dependencies>
+        <executions>
+          <execution>
+            <id>validate</id>
+            <phase>validate</phase>
+            <goals>
+              <goal>check</goal>
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <configLocation>google_checks.xml</configLocation>
+          <encoding>UTF-8</encoding>
+          <consoleOutput>true</consoleOutput>
+          <failsOnError>${checkstyle.failOnViolation}</failsOnError>
+          <failOnViolation>${checkstyle.failOnViolation}</failOnViolation>
+          <violationSeverity>warning</violationSeverity>
+          <format>xml</format>
+          <format>html</format>
+          <outputFile>${project.build.directory}/test/checkstyle-errors.xml</outputFile>
+          <linkXRef>false</linkXRef>
+        </configuration>
+      </plugin>
+    </plugins>
+
+    <pluginManagement>
+      <plugins>
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
-          <artifactId>maven-checkstyle-plugin</artifactId>
-          <version>2.17</version>
-          <dependencies>
-            <dependency>
-              <groupId>com.puppycrawl.tools</groupId>
-              <artifactId>checkstyle</artifactId>
-              <version>6.19</version>
-            </dependency>
-            <dependency>
-              <groupId>com.google.guava</groupId>
-              <artifactId>guava</artifactId>
-              <version>${dep.guava.version}</version>
-            </dependency>
-            <dependency>
-                <groupId>org.slf4j</groupId>
-                <artifactId>jcl-over-slf4j</artifactId>
-                <version>1.7.5</version>
-            </dependency>
-          </dependencies>
+          <artifactId>maven-dependency-plugin</artifactId>
+          <version>3.0.1</version>
           <executions>
             <execution>
-              <id>validate</id>
-              <phase>validate</phase>
+              <id>analyze</id>
```

Comment:
that was actually my very first approach as well but the ```arrow-format``` is already using the ```maven-dependency-plugin``` and its goal ```copy``` to fetch the flatc.exe. if I run the dependency plugin in the parent pom (with the ```analyze-only``` goal) then it shadows the ```copy``` goal instance in the ```arrow-format``` child.

I was trying to configure it so that the ```arrow-format``` is just extending the configuration adding/merging the ```copy``` goal execution with the inherited ```analyze-only``` global execution but had no luck with any sensible approach (it is only possible to reduce/activate some executions in child pom by overriding its phase to/from ```none``` but not adding completely new one; so the only possible way would be defining the plugin in the root pom with both ```analyze-only``` execution (for global dependency enforcements as per tis subj) and also the ```copy``` execution (ie with phase being ```none``` so that it can be used in the ```arrow-format``` child (setting the phase only in that module to ```generate-source```) - but specifying the ```copy-flatc``` execution in the parent pom would be ugly, hence my final approach with defining the ```analyze-only``` execution in ```pluginManagement``` and letting each module to instantiate (and override) it  itself).

so if you know a way to move it to the parent while having ```arrow-format``` still pulling the flatc.exe correctly that would still be my preferred way also...